### PR TITLE
Plotting satellite tracks

### PR DESCRIPTION
**Pull request by HF7weatherman**
This pull requests adds the capability to easily plot the forecasted satellite tracks. It further already uses this capability in the existing ICWV plots.

### CODE REVIEW

File: src/wblib/figures/internal/_general_plotting_functions.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import cartopy.crs as ccrs
+import numpy as np
+import pandas as pd
+import xarray as xr
+
+import orcestra.sat
+
+def plot_sattrack(valid_time, ax, satellite:str = "EARTHCARE"):
+    valid_day = str(valid_time.date())
+    sattracks = orcestra.sat.SattrackLoader(satellite, "2024-07-22"
```
Comment (d70-t) [inline src/wblib/figures/internal/_general_plotting_functions.py:11]: This is potentially dangerous:
The function `plot_sattrack` silently adds the forecast issuing date as an (unchangable) default argument. Any particular satellite track forecast will be outdated and we should **not** do flight planning on outdated and thus possibly wrong satellite track forecasts.

Comment (HF7weatherman) [reply src/wblib/figures/internal/_general_plotting_functions.py:11]: Yes, I agree. Thanks for the hint! I'll change that.

Comment (sortega87) [reply src/wblib/figures/internal/_general_plotting_functions.py:11]: Thanks for the changes so far!

How are you thinking to change the default argument? Right now I think we should provide it as users through "wbcli", and change it when we receive the message that it was updated. Perhaps we can also include a configuration file where we keep this file.

In any case I think we should afterwards store this date in the variables yaml file of each briefing and get it from a function within the wbcli library.

I can help with this if we agree is a good idea.

Comment (d70-t) [reply src/wblib/figures/internal/_general_plotting_functions.py:11]: Whatever fits the current workflow with `wbcli` best. My only concern is, that everyone using the satellite track predictions should in most cases explicitly be aware about which forecast has been used (so maybe an argument to `wbcli` is a good idea). Maybe it's also a good idea to print the forecast day of the satellite track somewhere on the plot.

Comment (sortega87) [reply src/wblib/figures/internal/_general_plotting_functions.py:11]: Thanks! I will work on the changes to the repo then. Making the date as an input argument for `wbcli`. The easiest place to save the track date is the quarto yaml file of each briefing, which can then be used to include the date in each presentation. We could also included the dates in the figures, but would need a change in the interface of the functions that generate the figures, so I would not recommend it in the moment. I will start including them in the yaml file.

Comment (sortega87) [reply src/wblib/figures/internal/_general_plotting_functions.py]: Actually had an error in my reasoning above, we will need to pass the track date to the figures. I will include it in the figures.

**Comment by sortega87** [comment]
Added now the satellite track forecast time as an input to `wbcli varaibles`. Updated the README to include a brief tutorial. Also included the date as an annotation.

**Comment by sortega87** [comment]
Here is a preview of the figure:

image

@HF7weatherman, I will stop now working on the branch so that you can include the latest changes that might be needed and merge it back to main.

**Comment by sortega87** [comment]
> Here is a preview of the figure:
> 
> image
> 
> @HF7weatherman, I will stop now working on the branch so that you can include the latest changes that might be needed and merge it back to main.

As discussed, also think we can merge

**Review by sortega87 — APPROVED** [review]

**Comment by sortega87** [comment]
@HF7weatherman I will do one more change before we commit. Make the `sattrack_time` an optional function for the internal functions.

**Comment by sortega87** [comment]
@HF7weatherman, @dspraturi, @Lucalino note that this pull request implies an interface change on the internal functions, going from 

`
def internal_function(briefing_time: pd.Timestamp, lead_hours: str) -> Figure:
    ...
` 

to

`
def internal_function(briefing_time: pd.Timestamp, lead_hours: str, sattrack_time: pd.Timestamp) -> Figure:
    ...
` 

At the moment there is only one figure, but new figures must adapt to the interface.

**Comment by HF7weatherman** [comment]
Allright, so we may also address #33 with this pull request. But I can only implement such things on Monday as I have to leave now. Have a nice weekend!

**Comment by sortega87** [comment]
I agree, sounds good. Thanks Henning! See you on Monday,

**Review by sortega87 — CHANGES_REQUESTED** [review]
Hi @HF7weatherman, I will review again after your changes. Let me know when I should take a look.

File: README.md [hunk 2]
```diff
@@ -16,10 +16,10 @@ where [BRIEFING_DATE] has the format "YYYYMMDD". This will create a new briefing
 Create the expected variable file in the folder with
 
 ```
-./wbcli.sh variables [BRIEFING_DATE] [FLIGHT_NUMBER] [LOCATION]
+./wbcli.sh variables [BRIEFING_DATE] [FLIGHT_CODE] [LOCATION] [SATTRACK_DATE]
```
Comment (Lucalino) [inline README.md:19]: Very minor but: Write SATTRACKS_DATE (plural on sattracks) to be consistent with variable used in code.

File: src/wblib/api/images.py [hunk 3]
```diff
@@ -27,7 +27,12 @@ def make_briefing_images(date: str, logger: Callable = logger) -> None:
     # internal
     internal_time = pd.Timestamp(date, tz=TIME_ZONE_STR)
     logger(f"Internal figure time set to {internal_time}", "INFO")
-    internal_figures = generate_internal_figures(internal_time, logger)
+    sattrack_time = pd.Timestamp(variables_dict["sattracks_date"],
```
Comment (Lucalino) [inline src/wblib/api/images.py:30]: Here it's singular again. Either one is fine, just be consistent. :)

File: src/wblib/api/images.py [hunk 4]
```diff
@@ -27,7 +27,12 @@ def make_briefing_images(date: str, logger: Callable = logger) -> None:
...
+                                 tz=TIME_ZONE_STR)
+    logger(f"Satellite forecasted tracks issued on {sattrack_time}", "INFO")
+    internal_figures = generate_internal_figures(internal_time,
```
Comment (Lucalino) [inline src/wblib/api/images.py:33]: same here

File: src/wblib/figures/internal/icwv.py [hunk 5]
```diff
@@ -9,9 +9,9 @@
 
 import matplotlib
 from matplotlib.figure import Figure
-
 import seaborn as sns
 
+from wblib.figures.sattrack import plot_sattrack
```
Comment (Lucalino) [inline src/wblib/figures/internal/icwv.py:14]: same here (-> sattracks)

File: src/wblib/figures/sattrack.py
Comment (Lucalino) [inline src/wblib/figures/sattrack.py:1]: Now, I am wondering if there was a logic behind sometimes using sattrack and other times sattracks.

**Comment by HF7weatherman** [comment]
@sortega87 All my changes are now added. If you approve the MR, I'll merge.

**Comment by HF7weatherman** [comment]
@sortega87 Sorry for being to fast! I'll add the plotting of the sattracks to all figures in this MR as well, probably a review after that would make more sense. I'll let you know once I'm finished.

**Comment by HF7weatherman** [comment]
@sortega87 Now this MR is ready for review.

**Comment by HF7weatherman** [comment]
Hey @sortega87 I resolved all merge conflicts and tested the result using `./wbcli`. For me, it's all working, so you could start with the review. Thanks in advance!

File: src/wblib/figures/briefing_info.py [hunk 6]
```diff
@@ -36,10 +37,13 @@ def format_internal_figure_axes(
...
     ax.set_ylim([lat_min, lat_max])
-    annotation = f"Latest ECMWF IFS forecast initialization: {issue_time.strftime('%Y-%m-%d %H:%M %Z')}"
+    annotation = (f"Latest ECMWF IFS forecast initialization: " +
+                  f"{issue_time.strftime('%Y-%m-%d %H:%M %Z')}\n" +
+                  f"Satellite tracks forecast issued on: "
+                  f"{sattracks_fc_time.strftime('%Y-%m-%d %H:%M %Z')}")
     ax.annotate(
         annotation,
-        (-26, -9),
+        (-29, -9),
```
Comment (sortega87) [inline src/wblib/figures/briefing_info.py:46]: Leave at 26 for better placement of the annotation.

File: src/wblib/figures/internal/icwv.py [hunk 7]
```diff
@@ -9,14 +9,14 @@
...
-
 import seaborn as sns
 
 from wblib.figures.briefing_info import INTERNAL_FIGURE_SIZE
 from wblib.figures.briefing_info import format_internal_figure_axes
 from wblib.figures.hifs import HifsForecasts
+from wblib.figures.sattrack import plot_sattrack
 
-
+CATALOG_URL = "https://tcodata.mpimet.mpg.de/internal.yaml"
```
Comment (sortega87) [inline src/wblib/figures/internal/icwv.py:19]: This line is not needed anymore

File: src/wblib/figures/briefing_info.py [hunk 8]
```diff
@@ -36,10 +37,13 @@ def format_internal_figure_axes(
     ax.set_xlabel("Longitude - \N{DEGREE SIGN}E")
     ax.set_xlim([lon_min, lon_max])
     ax.set_ylim([lat_min, lat_max])
-    annotation = f"Latest ECMWF IFS forecast initialization: {issue_time.strftime('%Y-%m-%d %H:%M %Z')}"
+    annotation = (f"Latest ECMWF IFS forecast initialization: " +
+                  f"{issue_time.strftime('%Y-%m-%d %H:%M %Z')}\n" +
+                  f"Satellite tracks forecast issued on: "
```
Comment (sortega87) [inline src/wblib/figures/briefing_info.py:42]: Include this information on the figure labels instead of the annotation.

**Review by sortega87 — CHANGES_REQUESTED** [review]
Hi @HF7weatherman! Thanks for the pull request. I had a couple of minor comments on the code and we can merge after addressing them.

**Pull request merged**